arch/arc/kernel/Makefile

```diff
@@ -13,6 +13,24 @@ else
 obj-y	+= intc-arcv2.o
 endif
 
+# Currently ftrace only works on arcv3 due to ABI
+ifdef CONFIG_ISA_ARCV3
+
+ifdef CONFIG_FTRACE
+# ftrace cant be traced (infinite loop)
+CFLAGS_REMOVE_ftrace.o = -pg
+CFLAGS_REMOVE_mcount.o = -pg
+
+# stack pointer changes midway through the function
```

Comment:
Why? Because we're switching to another task, I guess. Let's explain it here clearly.

Reply:
Great catch! After looking a bit more into this I realized it is no longer necessary, because I am no longer naively storing the stack pointer in the stack when performing "SAVE_ABI". Removed these two lines!